### ISSUE TITLE
non letter characters not triggering filterOption

### ISSUE THREAD

**Issue by madeleinehuish** [guiyep/react-select-virtualized#20]
I'm trying to search numerical strings in the value but the filterOptions callback isn't triggering on the input. We are trying to filter on UPC codes. We attempted to cast the value to a string but it is ignored in this library (it works fine in react-select).

**Comment by guiyep** [comment]
Thanks for reporting the issue.

Can you provide some examples?

Thanks!

**Comment by madeleinehuish** [comment]
Hi. So we have a large data set of about 1900 that we want to filter quickly with typeahead functionality. We have been using the react-select wrapped by our own hoc which works fine but it is too slow. We have to be able to filter on an item number as well as an item description but only show the item description in the input field. So we've been concatenating the number to the description in the value and passing that into the params for the function passed to filterOption functions. Here are the params we are passing: { { value: {concatenated string}, label: {what we want to display) } , current (which seems to be the current input value) }   . When we log current in the function passed to filterOption, it consoles out every time EXCEPT for when the string begins with a number. In the older version of react-select we currently have, there is an optional prop called onInputChange, which doesn't appear to be functioning with react-select-virtualized. If we could get that working, that could be another work around as then we could compare the value returned from onInputChange with the current value in the function passed to filterOption. I hope that all makes sense...

**Comment by guiyep** [comment]
Yep,

Ok I will take a look.

Can you in the meantime build a codesandox example? That would be very helpful.

Thanks!

**Comment by madeleinehuish** [comment]
Thanks! I'll see what I can put together.

**Comment by madeleinehuish** [comment]
So i've put together a sandbox here: https://codesandbox.io/s/sweet-shadow-m0qod.
On trying a smaller data set, it appears typing first with a number works correctly. However, when you type 1 in the field with the large data set, it shows no options... Please let me know if you see something...

**Comment by guiyep** [comment]
I think that you forgot to upload your code. I dont see any react-select in the sandbox you shared. onyl that data array objects,

**Comment by guiyep** [comment]
madeleinehuish ,

I think that I undestood now your particular issue. The  problem here in terms of this library is that I dont expose the filterOptions, since it affects a lot how you filter the option in how performant the virtualized list will load. I'm doing some optimizations to make the filtering more effective, but only by text.

I will think about this case.

On the other hand you can use the Async component. with thatone you will be in charge of loading the data and filtering.

Please check:
https://serene-hawking-021d7a.netlify.com/?path=/story/react-select-virtualized-async--async-with-default-options

Let me know how it goes.

Guiyep

**Comment by guiyep** [comment]
I added the filterOptions in the latest version. 2.0.25. 

Please check here: https://serene-hawking-021d7a.netlify.com/?path=/story/react-select-virtualized-props--with-filteroptions-value-with-number

thanks!

**Comment by madeleinehuish** [comment]
Hi, I’m away until Wednesday on a short bteak but will check first thing
when I get back. Thanks so much!

**Comment by madeleinehuish** [comment]
Hi, i just updated to 2.0.25 and it works like a charm :). Thank you SO much.

**Comment by JanBN** [comment]
Hello,

I have problem with filterOption. It is not called at all. 
This is my code:

```
            <Select
              name="form-field-name"
              value={this._location)}
              formatOptionLabel={(o) => o.computedName || ""}
              getOptionLabel={(o) => o.computedName || ""}
              getOptionValue={(o) => o.locationId}              
              options={this._locations}
              isClearable={true}
              filterOption={() => console.log('CALL MEEE !!')}
              isSearchable={true}

            />
```

I have also problem with your storybook example. I cant not see how it is done (click show info) because my chrome just freezes. Could you please update storybook with just a few items.

thank you

**Comment by guiyep** [comment]
It is filterOptions not filterOption. Check the story.jsx file. The code example that appears in storybook it is pretty basic, it is being auto generated by a storybook plugin.

**Comment by JanBN** [comment]
Thank you for the quick response. Unfortunately not even filterOptions works for me.
Are you sure it should be filterOptions ? Could you please double check it. I have searched the code and for example here: https://github.com/guiyep/react-select-virtualized/blob/50954195af329358d7fec24f68f9ad37615db38f/src/select-virtualized/components/select/Select.stories.jsx#L49 is filterOption

Also in original react select there is filterOption - https://react-select.com/props

I'm using this version:
**"react-select-virtualized": "^2.3.0",**

Thank you

**Comment by guiyep** [comment]
Sorry you are right. Strange that storybook works and not in your case.

I will have to take a look, sorry.

**Comment by JanBN** [comment]
thank you, please let me know if you find out something.

**Comment by guiyep** [comment]
I will reopen this issue.

**Comment by guiyep** [comment]
@JanBN  just as an update, there is an issue with babel that I'm trying to fix. I will let you know as soon as I fix it.

**Comment by guiyep** [comment]
@JanBN this wise fixed in #2.3.3, It has being pushed need to wait for npm to pick up the changes.
